docs/designs/cre-links.md

```diff
@@ -0,0 +1,55 @@
+# CRE Links
+
+The biggest advantage of opencre.org is its ability to dynamically update itself by parsing other standards that link to it.
+So far, as a small project, our approach to this has been relatively simple: when any security resource added a link to opencre.org in their documents we would write a parser for their specific document format and information structure and add it to an ever-growing list of custom parsers.
+
+This works relatively well for big, well-structured projects that have sufficient automation to either release big spreadsheets (e.g. CCM) or several simply structured Markdown files with one topic per section/file (Cheatsheets, ASVS etc).
+
+However, it fails for single-page documents with several sections or information and for mapping lots of small standards which will likely have their own unique structure each.
+
+
+## The problem
+
+Human-readable documents can have several structures and links can have even more, parsing and understanding human language is an open problem in computing.
+For example, there are likely non-NLP/AI ways to extract the correct link and title from a markdown file structured like this.
+
+``` Markdown
+# Document Title
+
+## Relevant Title
+[...]
+
+## References
+https://opencre.org/cre/<>
+
+```
+We could make an assumption that we always pick the document title as the name for each document that links to us but then what happens if we get 2 cre links for different purposes? A markdown files could contain multiple sections of information.
+
+In general parsing this kind of human readable information, complicates the project beyond its scope and maintaining those solutions requires significant resources.
+
+= Suggested solution=
+Since standards writers are already required to provide links to opencre.org why not get them to encode all the information we need in the link?
```

Comment:
```suggestion

As standards integrate with the OpenCRE.org project, it would be intuitive for the authors to inject the information needed for the project to function with its maximum potential.
```

Reply:
See earlier comment. The alternative is that this is done just once for a file or for a project, in a configuration comment, or in a separate configuration file for the opencre parser, containing standard name, version,  the grep patterns to search for the correct metadata and the pattern to construct the hyperlink.